torch_xla/csrc/ops/generic.cpp

```diff
@@ -37,11 +38,48 @@ Generic::Generic(torch::lazy::OpKind op, xla::Shape shape, LowerFn lower_fn,
       lower_fn_(std::move(lower_fn)),
       hash_seed_(hash_seed) {}
 
+Generic::Generic(torch::lazy::OpKind op, torch::lazy::OpList operands,
+                 xla::Shape xla_shape, LowerFn lower_fn,
+                 xla::OpSharding sharding, size_t num_outputs,
+                 torch::lazy::hash_t hash_seed)
+    : XlaNode(std::move(op), operands,
+              {XlaHelpers::ConvertXlaShapeToLazy(xla_shape)},
+              std::move(xla_shape), sharding, num_outputs, hash_seed),
+      lower_fn_(std::move(lower_fn)),
+      hash_seed_(hash_seed) {}
+
+torch::lazy::NodePtr Generic::Clone() const {
+  std::optional<torch::lazy::OpList> ops = oplist();
+  if (ops.has_value()) {
+    return Clone(ops.value());
+  } else {
+    return torch::lazy::MakeNode<Generic>(op(), xla_shape(), lower_fn_,
+                                          num_outputs(), hash_seed_);
+  }
+}
+
 torch::lazy::NodePtr Generic::Clone(torch::lazy::OpList operands) const {
   return torch::lazy::MakeNode<Generic>(op(), operands, xla_shape(), lower_fn_,
                                         num_outputs(), hash_seed_);
 }
 
+torch::lazy::NodePtr Generic::CloneWithSharding(
+    xla::OpSharding sharding) const {
+  // TODO(steventk) Right now, we drop the operands on clone, because the oplist
+  // memory becomes unsafe when we clone the other nodes and they go out of
+  // scope. Instead of initializing the oplist below to nullopt, we want to use
```

Comment:
Thiking out loud, constructing the `oplist` from the parent's `operands_`, if possible, could simplify things. As long as the operands are alive, they shoulnd't go out of scope. For the recreated node, we should propagate that change (old->new), though.

Reply:
> constructing the oplist from the parent's operands_

Ah, like "clone" the operand in the oplist instead of the parent. That makes sense to me, but I'm not sure how to get a reference to the parent